app/plugins/adp-backend/src/plugin.ts

```diff
@@ -0,0 +1,142 @@
+import {
+  coreServices,
+  createBackendPlugin,
+} from '@backstage/backend-plugin-api';
+import { fetchApiRef } from '@internal/plugin-fetch-api-backend';
+import { ArmsLengthBodyStore } from './armsLengthBody';
+import { DeliveryProjectStore, FluxConfigApi } from './deliveryProject';
+import { DeliveryProgrammeStore } from './deliveryProgramme';
+import { DeliveryProgrammeAdminStore } from './deliveryProgrammeAdmin';
+import { DeliveryProjectUserStore } from './deliveryProjectUser';
+import {
+  DeliveryProjectGithubTeamsSyncronizer,
+  GitHubTeamsApi,
+  GithubTeamStore,
+} from './githubTeam';
+import { DefaultIdentityClient } from '@backstage/plugin-auth-node';
+import { CatalogClient } from '@backstage/catalog-client';
+import { DeliveryProjectEntraIdGroupsSyncronizer, EntraIdApi } from './entraId';
+import {
+  createAlbRouter,
+  createDeliveryProgrammeAdminRouter,
+  createDeliveryProjectUserRouter,
+  createProgrammeRouter,
+  createProjectRouter,
+} from './service';
+import { Router } from 'express';
+import { initializeAdpDatabase } from './database';
+
+export const adpPlugin = createBackendPlugin({
+  pluginId: 'adp',
+  register(env) {
+    env.registerInit({
+      deps: {
+        logger: coreServices.logger,
+        discovery: coreServices.discovery,
+        database: coreServices.database,
+        config: coreServices.rootConfig,
+        permissions: coreServices.permissions,
+        fetchApi: fetchApiRef,
+        httpRouter: coreServices.httpRouter,
+      },
+      async init({
+        logger,
+        discovery,
+        database,
+        config,
+        permissions,
+        fetchApi,
+        httpRouter,
+      }) {
+        await initializeAdpDatabase(database);
+
+        const dbClient = await database.getClient();
```

Comment:
Could all these service initializations be moved to somewhere else perhaps? A lot of them could be registered as services and injected in I think. Maybe something for a future PR?